### DISCUSSION TITLE
Welcome to react-complex-tree Discussions!

### DISCUSSION THREAD

**Discussion by lukasbach** [lukasbach/react-complex-tree#11] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by jorgeanjos** [comment]
Hello.
Can you help me?
I am trying to implement the onDidChangeTreeData on the dataProvider in an UncontrolledTreeEnvironment but I dont know how and I dont see any examples of it. I want the tree data to refresh when I click on a element of the tree.

dataProvider={{
              onDidChangeTreeData: // ? what do I do ?
              getTreeItem: async (itemId) => {
                return new Promise((res) => res(listItems[itemId]));
              },
            }}

**Reply by lukasbach** [reply]
Hi @jorgeanjos! There are essentially two ways how tree data can be changed, either by moving items to a different location (which will invoke `dataProvider.onChangeItemChildren()`) or by renaming an item (which will invoke `dataProvider.onRenameItem()`). Please use these methods to get informed of data changes. The `onDidChangeTreeData` prop is a event emitter that the data provider can optionally implement to propagate change events to outside, which mostly is used by the `StaticTreeDataProvider` to allow external consumers to register change events. I admit that the naming here is suboptimal, as `onDidChangeTreeData` is not an event itself but an emitter, maybe that should be renamed..

**Comment by dunnkers** [comment]
This repo deserves more stars! Great job on the package, works flawlessly! 🥳
